src/frontend/Sider/Behandling/Inngangsvilkår/Stønadsperioder/RedigerStønadsperioder.tsx

```diff
@@ -0,0 +1,173 @@
+import { useBehandling } from '../../../../context/BehandlingContext';
+import { useInngangsvilkår } from '../../../../context/InngangsvilkårContext';
+import useFormState, { FormErrors, FormState } from '../../../../hooks/felles/useFormState';
+import { ListState } from '../../../../hooks/felles/useListState';
+import { Feilmelding } from '../../../../komponenter/Feil/Feilmelding';
+import { RessursStatus } from '../../../../typer/ressurs';
+import { leggTilTomRadUnderIListe } from '../../VedtakOgBeregning/Barnetilsyn/utils';
+import { Stønadsperiode, Vilkårperioder } from '../typer';
+
+const Grid = styled.div`
+    display: grid;
+    grid-template-columns: repeat(5, max-content);
+    grid-gap: 0.5rem 1rem;
+    align-items: start;
+
+    > :nth-child(5n) {
+        grid-column: 1;
+    }
+`;
+
+const Knapp = styled(Button)`
+    max-width: fit-content;
+    margin-top: 1rem;
+`;
+
+export type StønadsperiodeForm = {
+    stønadsperioder: Stønadsperiode[];
+};
+const tomStønadsperiodeRad = (): Stønadsperiode => ({
+    målgruppe: '',
+    aktivitet: '',
+    fom: '',
+    tom: '',
+});
+
+const initFormState = (
+    eksisterendeStønadsperioder: Stønadsperiode[]
+): FormState<StønadsperiodeForm> => ({
+    stønadsperioder:
+        eksisterendeStønadsperioder.length !== 0
+            ? eksisterendeStønadsperioder
+            : [tomStønadsperiodeRad()],
+});
+
+const RedigerStønadsperioder: React.FC<{
+    vilkårperioder: Vilkårperioder;
+    eksisterendeStønadsperioder: Stønadsperiode[];
+}> = ({ vilkårperioder, eksisterendeStønadsperioder }) => {
+    const { oppdaterStønadsperioder } = useInngangsvilkår();
+    const { request } = useApp();
+    const { behandling } = useBehandling();
+
+    const [feilmelding, settFeilmelding] = useState<string>();
+    const [laster, settLaster] = useState<boolean>(false);
+    const validerForm = (formState: StønadsperiodeForm): FormErrors<StønadsperiodeForm> => {
+        return {
+            stønadsperioder: validerStønadsperioder(
+                formState.stønadsperioder,
+                vilkårperioder.målgrupper,
+                vilkårperioder.aktiviteter
+            ),
+        };
+    };
+    const formState = useFormState<StønadsperiodeForm>(
+        initFormState(eksisterendeStønadsperioder),
+        validerForm
+    );
+
+    const stønadsperioderState = formState.getProps('stønadsperioder') as ListState<Stønadsperiode>;
+
+    const handleSubmit = (form: FormState<StønadsperiodeForm>) => {
+        if (laster) return;
+        settLaster(true);
+        settFeilmelding(undefined);
+        return request<Stønadsperiode[], Stønadsperiode>(
+            `/api/sak/stonadsperiode/${behandling.id}`,
+            'POST',
+            form.stønadsperioder
+        )
+            .then((res) => {
+                if (res.status === RessursStatus.SUKSESS) {
+                    stønadsperioderState.setValue(res.data);
+                    oppdaterStønadsperioder(res);
```

Comment:
Usikker på om dette burde skje i en if-sjekk eller om man skulle flyttet hele funksjonen ut i context? Fordelen er at nå lever feilmeldingen her inne. 

Trenger man fortsatt å oppdatere state? Eller kan den fjernes? Må man kanskje slette formstate? Eller skjer det automatisk ved at den ikke rendres lenger?

Reply:
Det som kanskje blir litt rart i denne flyten nå er at man har en state i en context, som egentlige kan oppdateres utenfor denne komponenten (selv om det ikke skjer). Samt en state inne i denne komponenten som er beroende på context-statet. Som burde oppdateres hvis staten fra params blir oppdatert.

Inne i handleSubmit oppdaterer man begge states. 

Et alternativ til dette er å kun oppdatere stønadsperioder her, med ``oppdaterStønadsperioder(res);` og legge til en `useEffect` for å oppdatere statet for komponenten
```typescript
useEffectNotInitialRender(() => {
        stønadsperioderState.setValue(eksisterendeStønadsperioder);
        formState.setErrors({
            stønadsperioder: validerStønadsperioder(
                eksisterendeStønadsperioder,
                vilkårperioder.målgrupper,
                vilkårperioder.aktiviteter
            ),
        });
    }, [eksisterendeStønadsperioder]);
```